### PR TITLE
Add linter job in the CI

### PR DESCRIPTION
**Pull request by PtitLuca**
## Description

This PR adds a job in the continuous integration which runs the linter.
It also provides an empty TypeScript project.

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have tested this code
- [ ] I have added sufficient documentation in the code

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  pull_request:
+    branches: ["main", "master", "dev"]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
```
Comment (TomChv) [inline .github/workflows/ci.yaml:15]: Add a real name
`name: Lint`

File: .prettierrc.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  semi: true,
+  trailingComma: "all",
+  singleQuote: true,
+  printWidth: 120,
+  tabWidth: 4
```
Comment (TomChv) [inline .prettierrc.js:6]: Add some rules

```
bracketSpacing: true,
useTabs: true
```

Comment (TomChv) [reply .prettierrc.js:6]: Config tabs to 2, it's a backend project

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    // "noUncheckedIndexedAccess": true,                 /* Include 'undefined' in index signature results */
+    // "noImplicitOverride": true,                       /* Ensure overriding members in derived classes are marked with an override modifier. */
+    // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type */
+    // "allowUnusedLabels": true,                        /* Disable error reporting for unused labels. */
+    // "allowUnreachableCode": true,                     /* Disable error reporting for unreachable code. */
+
+    /* Completeness */
+    // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
+    "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
+  }
```
Comment (TomChv) [inline tsconfig.json:100]: You forget include and exclude options in your `tsconfig.json`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "main": "src/index.ts",
+  "scripts": {
+    "build": "tsc",
+    "start": "node ./dist/index.js",
+    "dev": "ts-node ./src/index.ts",
+    "lint": "eslint '*/**/*.{js,ts,tsx}'",
+    "lint:fix": "eslint '*/**/*.{js,ts,tsx}' --fix"
+  },
+  "repository": "git@github.com:PtitLuca/scaleway-sdk-js.git",
+  "author": "Luca Georges Francois <luca.georges-francois@epitech.eu>",
```
Comment (TomChv) [inline package.json:14]: And me ? :(

**Review by TomChv — CHANGES_REQUESTED** [review]
LGTM, just fix some of comment

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (PtitLuca) [reply .github/workflows/ci.yaml:16]: Added in https://github.com/PtitLuca/scaleway-sdk-js/pull/1/commits/ebd1af2150ebc35ebd9015656aa4b17b156e165f.

File: .prettierrc.js (hunk 2 above)
Comment (PtitLuca) [reply .prettierrc.js]: Added both in https://github.com/PtitLuca/scaleway-sdk-js/pull/1/commits/ebd1af2150ebc35ebd9015656aa4b17b156e165f.

File: tsconfig.json (hunk 3 above)
Comment (PtitLuca) [reply tsconfig.json]: Added in https://github.com/PtitLuca/scaleway-sdk-js/pull/1/commits/ebd1af2150ebc35ebd9015656aa4b17b156e165f.

File: package.json (hunk 4 above)
Comment (PtitLuca) [reply package.json:14]: Added in https://github.com/PtitLuca/scaleway-sdk-js/pull/1/commits/ebd1af2150ebc35ebd9015656aa4b17b156e165f.

**Pull request merged**